### PR TITLE
Adds a GraphQL Endpoint

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

**Comment by chirino** [comment]
Still a wip, this is just a rebase of some previous work I had done to add a GraphQL endpoint.  Basically I need update the UI to actually do something useful with it

File: app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+public class GraphQLConfiguration {
+
+    @Autowired
+    IntegrationHandler integrationHandler;
+
+    @Bean
+    GraphQLSchema schema() throws IOException {
+
+        String schema;
+        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("schema.graphqls") )  {
```
Comment (zregvart) [inline app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java]:
```suggestion
        try (InputStream is = GraphQLConfiguration.class.getResourceAsStream("schema.graphqls") )  {
```

I think we have utility method that loads from classpath, lookup `Resources` utility class.

File: app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        TypeDefinitionRegistry typeDefinitionRegistry = schemaParser.parse(schema);
+
+        SchemaGenerator schemaGenerator = new SchemaGenerator();
+        return schemaGenerator.makeExecutableSchema(typeDefinitionRegistry, newRuntimeWiring()
+
+            // Entities who's json shape matches that GraphQL schema,
+            // just need a a bit of code like this to load the data for the query.
+            .type("Query", builder -> builder
+                .dataFetcher("integrations", env->{
+                    return integrationHandler.list(null);
```
Comment (zregvart) [inline app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java]: I'd avoid this coupling of handler to a handler, we have already done this and it has caused only issues.

File: app/pom.xml [hunk 3]
```diff
@@ -542,22 +546,28 @@
...
-                <arg>-XepExcludedPaths:.*/generated-sources/.*</arg>
-                <arg>-implicit:none</arg>
-                <!-- temporary disable until https://github.com/google/error-prone/issues/780 is resolved -->
-                <arg>-Xep:ParameterName:OFF</arg>
-              </compilerArgs>
-            </configuration>
+            <executions>
+              <execution>
+                <id>default-compile</id>
+                <phase>compile</phase>
```
[the remark this answers is not in the document]
Comment (chirino) [reply app/pom.xml]: thats exactly the point.  This change disable the lint checks on test cases.

File: app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/operations/PaginationOptionsFromQueryParams.java [hunk 4]
```diff
@@ -33,19 +33,24 @@
      * @param uri The request context.
      */
     public PaginationOptionsFromQueryParams(UriInfo uri) {
-        MultivaluedMap<String, String> queryParams = uri.getQueryParameters();
-        String pageQuery = queryParams.getFirst("page");
-        if (pageQuery == null || pageQuery.isEmpty()) {
+        if( uri == null ) {
```
[the remark this answers is not in the document]
Comment (chirino) [reply app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/operations/PaginationOptionsFromQueryParams.java]: This just default faults the value if URI is null which is the case when the graphql enpoint uses it.

File: app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java (hunk 1 above)
Comment (chirino) [reply app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java]: Cool.. I'll switch to that.

File: app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java (hunk 2 above)
Comment (chirino) [reply app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java]: can you point me at what those issues were?

**Comment by chirino** [comment]
This does have a unit test at: https://github.com/syndesisio/syndesis/pull/6208/files#diff-95425df74de456611e2f7ae014ad229dR88

File: app/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (zregvart) [reply app/pom.xml]: Than -1 on that. There's no reason for tests to be worst off that production code.

File: app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java (hunk 2 above)
Comment (zregvart) [reply app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java]: #5995 and #6111

**Comment by zregvart** [comment]
> This does have a unit test at: [syndesisio/syndesis/pull/6208/files#diff-95425df74de456611e2f7ae014ad229dR88](https://github.com/syndesisio/syndesis/pull/6208/files#diff-95425df74de456611e2f7ae014ad229dR88)

That would be an integrated test not a unit test, it starts the server and interacts with it via HTTP.

**Pull request closed without merging**